pkg/apis/nodeup/config.go

```diff
@@ -62,11 +64,23 @@ type Config struct {
 	// specified, each parameter must follow the form variable=value, the way
 	// it would appear in sysctl.conf.
 	SysctlParameters []string `json:",omitempty"`
+	// UpdatePolicy determines the policy for applying upgrades automatically.
+	UpdatePolicy string
 	// VolumeMounts are a collection of volume mounts.
 	VolumeMounts []kops.VolumeMountSpec `json:",omitempty"`
 
 	// ConfigServer holds the configuration for the configuration server
 	ConfigServer *ConfigServerOptions `json:"configServer,omitempty"`
+	// AuxConfigHash holds a secure hash of the nodeup.AuxConfig.
+	AuxConfigHash string
+}
+
+// AuxConfig is the configuration for the nodeup binary that might be too big to fit in userdata.
+type AuxConfig struct {
```

Comment:
Maybe instead of calling this AuxConfig, NodeFullConfig would be consistent with ClusterFullConfig.  I see, nodeup.Config as more the auxiliary configuration ... it is the small config (constrained by userdata size) that lets us locate and load the real config.

Reply:
I'd prefer not to put duplicate, unused fields in this object.

I have thought of using the same schema for userdata's `Config` and this file in the state store. That would allow putting the hooks, etc. in the userdata when they fit. The userdata would then have metadata saying "pull these fields from the one in the state store" when things don't fit.

But that is all more complicated and should be deferred until after we get more experience and better testing around the exceptional cases.